File: oidc/management/commands/seed_jenkins.py
# oidc/management/commands/seed_jenkins.py

from django.core.management.base import BaseCommand
from oidc.models import JenkinsClient
import secrets


class Command(BaseCommand):
    help = "Seed Jenkins client"

    def handle(self, *args, **kwargs):
        api_key = secrets.token_urlsafe(32)

        obj, created = JenkinsClient.objects.update_or_create(
            name="jenkins-prod",
            defaults={"api_key": api_key}
        )

        self.stdout.write(self.style.SUCCESS(f"API KEY: created={created}, api_key={api_key}"))

File: scripts/seedJenkins.js
import crypto from "node:crypto";
import { assertConfig } from "../src/config.js";
import { connectDb, disconnectDb } from "../src/db.js";
import { JenkinsClient } from "../src/models/JenkinsClient.js";

// Equivalent of the Django `seed_jenkins` management command: upsert the
// "jenkins-prod" client with a freshly generated URL-safe API key.
async function main() {
  assertConfig();
  await connectDb();

  const apiKey = crypto.randomBytes(32).toString("base64url");

  const result = await JenkinsClient.findOneAndUpdate(
    { name: "jenkins-prod" },
    { $set: { apiKey } },
    { new: true, upsert: true, rawResult: true }
  );

  const created = result.lastErrorObject?.updatedExisting === false;

  console.log(`API KEY: created=${created}, api_key=${apiKey}`);
}

main()
  .catch((err) => {
    console.error(err);
    process.exitCode = 1;
  })
  .finally(async () => {
    await disconnectDb();
  });
